elastic/kibana
# Deprecate Anonymous Authentication Credentials

Resolves #128877

### Overview

This PR adds deprecation warnings for 2 credential types of anonymous authentication providers:
- apiKey (this includes both key and id/key pair)
- 'elasticsearch_anonymous_user'

Any configured anonymous authentication providers with either of these credentials will trigger a depreciation warning, even if the provider is not enabled.

This PR also adds usage telemetry for the credential type of the enabled anonymous authentication provider, defined as anonymousCredentialType, with the following possible values:
- undefined (anonymous authentication provider not configured or not enabled)
- api_key (either key or id/key pair)
- elasticsearch_anonymous_user (literal credential value 'elasticsearch_anonymous_user')
- username_password (username/password)

### Testing
1. [Modify elasticsearch.yaml to include an anonymous user ](https://www.elastic.co/guide/en/elasticsearch/reference/current/anonymous-access.html)
2. Start Elasticsearch.
3. [Modify kibana.yaml to include anonymous authentication](https://www.elastic.co/guide/en/kibana/current/kibana-authentication.html#anonymous-authentication)
4. Start Kibana.
6. Verify deprecation warning for apiKey and 'elasticsearch_anonymous_user' credentials. Verify no deprecation warning for username/password credentials.
8. Remove anonymous authentication from yaml files. Verify no deprecation warning for anonymous authentication.

Notes
A decision was made to include disabled anonymous authentication providers in the deprecation check. Usage telemetry is only concerned with the enabled anonymous authentication provider.
